### PR TITLE
Start tutorial showing how to use PyGMT with other packages

### PR DESCRIPTION
**Pull request by maxrjones**
Create the notebook and add it to the TOC.

Note, I might need to find a different dataset to use for the interactive component. It previously was working great, but now oddly all the depth levels seem to contain the same data.

Also, let me know if you think that an interactive part for this demo should be switched out for a simpler example.

### CODE REVIEW

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]: Some units for the colorbar would be nice here. I don't understand the number range though, it goes from 0 up to 0.14, is GDP per capita meant to be that low?

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb:241]: > Note, I might need to find a different dataset to use for the interactive component. It previously was working great, but now oddly all the depth levels seem to contain the same data.

I'm also having trouble getting the plot to change with the slider in my local Jupyter notebook. Are you sure it's the data, or something up with `panel`?

Another thing is with rendering the Panel plot in JupyterBook. There's supposed to be a way to get interactivity via `thebe` (see https://jupyterbook.org/en/stable/interactive/thebe.html), but not sure how it works in practice. You could play with the config and see how it goes (no rush though, you can do this in a follow-up PR after the recording).

P.S. I'm also trying to get `pygmt` working in `pyodide` at https://github.com/pyodide/pyodide/pull/2525, but don't count on that getting finished anytime soon.

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]: The attendees won't know about `grdimage` prior to this. Maybe you could have a static example of `grdimage` a few cells above first, and then move on to the interactive component?

File: book/ecosystem.ipynb
Comment (maxrjones) [reply book/ecosystem.ipynb:241]: Yes, I think so because np.testing.assert_allclose on the values from two depth slices didn't report any differences and I got the same result with a quick matplotlib plot of different depths. I'll try to figure it out tomorrow, but wanted to open the PR in case you and Leo disagreed with the general idea for the content.

> P.S. I'm also trying to get pygmt working in pyodide at https://github.com/pyodide/pyodide/pull/2525, but don't count on that getting finished anytime soon.

Awesome, super excited to see the outcome!

File: book/ecosystem.ipynb
Comment (maxrjones) [reply book/ecosystem.ipynb]: Good idea, I'll add that!

File: book/ecosystem.ipynb
Comment (maxrjones) [reply book/ecosystem.ipynb]: I'll update it to plot the world population column instead for simplicity.

File: book/ecosystem.ipynb
Comment (maxrjones) [reply book/ecosystem.ipynb:241]: Weirdly enough, it works fine when reading from disk as a netcdf. It's not a PyGMT/GMT problem, something to do with either the IRI store or OPeNDAP. I changed it to download using pygmt.which first.

File: book/ecosystem.ipynb
Comment (maxrjones) [reply book/ecosystem.ipynb]: I'm holding off on this comment to see if we'll have a separate tutorial for inset and/or subplot that includes grdimage.

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]: Show `cmap_bounds` variable which is `array([1.40000000e-10, 1.37930277e-03])`

```suggestion
    "cmap_bounds = pygmt.info(data=world['pop_est'], per_column=True)\n",
    "cmap_bounds"
```

File: environment.yml
Comment (weiji14) [inline environment.yml]: Somehow I had to update to panel=0.13.0 for the interactive plot to show up. Does it work for you with panel=0.12.4?
```suggestion
  - panel=0.13.0
```

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]: Again, good to show what the `woa_temp` xarray.Dataset looks like.
```suggestion
    "woa_temp = xr.open_dataset(netcdf_file)\n",
    "woa_temp"
```

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]: Minor suggestion, but was thinking whether the `.isel(time=0)` could be done in one of the cells above? It might make the rendering slightly faster to not have the do this slicing everytime the slider is dragged.

File: book/ecosystem.ipynb
Comment (weiji14) [inline book/ecosystem.ipynb]:
```suggestion
    "depth_slider = pn.widgets.DiscreteSlider(name='Depth (m)', options=woa_temp.depth.values.astype(int).tolist(), value=0)"
```

**Comment by maxrjones** [comment]
First cut at a recording of the tutorial - 

I would like to re-record this tomorrow AM to improve a few points, but am sharing this version so others can plan for timing. It's currently ~16 minutes and I do not expect the next recording to be any longer. Please ignore the first 15 seconds and the last few seconds, I would normally trim it but don't have time for the draft.

**Comment by leouieda** [comment]
@meghanrjones this is really cool! Learned a lot from your tutorial 🙂 I hadn't played around with panel before and never used the geopandas integration. Looking forward to trying these out for myself!

**Comment by weiji14** [comment]
Looks great! Not sure if you've recorded your 2nd one already, but just as general suggestion, you could go fullscreen and go set "View -> Presentation Mode" as suggested in https://github.com/GenericMappingTools/egu22pygmt/pull/5#issuecomment-1124678018 to make things a tiny bit bigger (though I was viewing this on my phone and it looked ok). Oh, and ~15-16 minutes is totally fine, we're running a bit long for some of the other ones so this can help with the time budget.

P.S. If you're happy with the notebook, just resolve the merge conflict with the main branch and let me know once it's ready to merge.

**Comment by maxrjones** [comment]
Here's a final version (let me know if you need a different sharing method). Thanks for the tip, I used the presentation mode. Yes, I think this is good to merge.

https://user-images.githubusercontent.com/14077947/168496029-b89be08e-32f8-4647-8fa8-fa68eac22da4.mp4

**Review by weiji14 — APPROVED** [review]

**Comment by weiji14** [comment]
Thanks again Max for making this tutorial! The notebook is now rendered at https://www.generic-mapping-tools.org/egu22pygmt/ecosystem.html. Everything looks ok ~~except for the panel plot in the last cell which isn't rendering on Jupyter Book. Not sure if there's a way to get it rendered like in the Panel gallery at https://panel.holoviz.org/reference/widgets/DiscreteSlider.html#display, but we can try and sort that out next week if there's time.~~ Edit: Sorry, just realized that I had the javascript disabled. It's actually displaying fine, just that it's not interactive.

image

Oh, and the video looks good to me, I'll spend some time in the next few days merging all of them together :smile:

**Comment by maxrjones** [comment]
Thanks Wei Ji for reviewing all the tutorials/videos and stitching them together!

**Comment by maxrjones** [comment]
FYI I'll look into whether the last visualization can be made interactive in jupyterbook later this week - it would be nice but isn't essential to have for the upload deadline.

**Pull request merged**
